### PR TITLE
Added test that verify Internet connection from inside of container.

### LINKED ISSUE #9662: E2E: Modify networking.go to verify containers (rather than client) can access internet.
From a recent thread https://groups.google.com/forum/#!topic/google-containers/aDdYFLcQxk4 , it looks  like we can modify networking.go to do a test of external connectivity which is actually useful :). 

The snippet im suggesting we replace ... 

```
resp, err := http.Get("http://google.com")
		if err != nil {
			Failf("Unable to connect/talk to the internet: %v", err)
		}
		if resp.StatusCode != http.StatusOK {
			Failf("Unexpected error code, expected 200, got, %v (%v)", resp.StatusCode, resp)
		}
```

... in its place, we should write a test w/ a  simple pod which pings google **FROM INSIDE A CONTAINER** and prints "SUCCESS" after its done, then grep for the SUCCESS string.

### PR DESCRIPTION
**Pull request by piosz**
Fixes #9662

### CODE REVIEW

**Comment by piosz** [comment]
cc @quinton-hoole @jayunit100 @thockin

**Comment by jayunit100** [comment]
Can we avoid manually deleting the pod ?  That kind of thing makes the tests harder to understand given that we rely on namespace lifecycles.

**Comment by quinton-hoole** [comment]
I'd like to review this before it merges please. I don't necessarily think that removing pod deletion from the test is a good thing. More later when I'm awake and not on my phone. :-)

File: test/e2e/networking.go [hunk 1]
```diff
@@ -50,6 +50,36 @@ var _ = Describe("Networking", func() {
...
+			},
+			ObjectMeta: api.ObjectMeta{
+				Name: podName,
+			},
+			Spec: api.PodSpec{
+				Containers: []api.Container{
+					{
+						Name:    contName,
+						Image:   "gcr.io/google_containers/busybox",
+						Command: []string{"ping", "-c", "5", "-w", "60", "google.com"},
```
Comment (thockin) [inline test/e2e/networking.go]: This is a pretty heavy timeout.  could it just be -c 3 -w 10 ?

**Comment by thockin** [comment]
Thanks for jumping on this

**Comment by thockin** [comment]
I agree this is not appropriate for air-gapped systems, but those systems are not generally running our e2e for now.

File: test/e2e/networking.go (hunk 1 above)
Comment (quinton-hoole) [inline test/e2e/networking.go]: Don't we want a TCP rather than an ICMP test here?

Comment (quinton-hoole) [inline test/e2e/networking.go]: PS:  ICMP is often dropped by routers, firewalls etc.

**Comment by quinton-hoole** [comment]
Regarding @jayunit100 's suggestion to omit the pod deletion, I'd prefer to leave it in, and use the blanket namespace deletion as a safety blanket, rather than the sole mechanism by which resources are cleaned up.  I should document a "best practises" guide around this and other things.

File: test/e2e/networking.go (hunk 1 above)
Comment (jayunit100) [inline test/e2e/networking.go]: Why not both?

**Comment by jayunit100** [comment]
Sorry to rathole on this manual deletion thing  but : why should we do something redundantly when we have a declarative cleanup functionalitythat already is known to work?   There is no benefit to the system and a small
cost to test readability which will possibly accumulate if others follow the same pattern of manually deleting everything .

**Comment by piosz** [comment]
@jayunit100 Actually most of the tests destroy all object manually and then cleanup manually. We may want change "best practises" as @quinton-hoole mentioned but let's leave it as for now to make all tests consistent.

@quinton-hoole are you ok with leaving ICMP? As @thockin wrote it should be fine for systems running e2e.

**Comment by quinton-hoole** [comment]
I'd rather use TCP, for two reasons:
1. TCP port 80 is always reachable on public HTTP servers, by definition.  ICMP is not.  Only by accident it sometimes is.  This is inviting false negatives.
2. Proving that google.com is accessible on ICMP is generally not actually useful.  So you're inviting false positives too.

It should be as simple as using netcat (nc), as in:

nc -z google.com 80

If you'd like to defer that to a later PR I'm ok with that.

**Comment by thockin** [comment]
TCP is better, but I'd be happy to take this as-is and change it tomorrow.

**Comment by piosz** [comment]
Changed timeout according to Tim's comment. Will create a follow up PR with TCP tomorrow.

**Comment by thockin** [comment]
LGTM with the promise of followup :)

**Comment by piosz** [comment]
Thanks!

**Comment by mbforbes** [comment]
Shippable...

**Comment by mbforbes** [comment]
Bah

**Comment by dchen1107** [comment]
Thanks a lot on creating this test.

**Pull request merged**